lib/dependencies.sh

```diff
@@ -305,3 +316,47 @@ npm_prune_devdependencies() {
     meta_set "skipped-prune" "false"
   fi
 }
+
+pnpm_install() {
+  local build_dir=${1:-}
+
+  echo "Running 'pnpm install' with pnpm-lock.yaml"
+  cd "$build_dir" || return
+
+  monitor "pnpm-install" pnpm install --prod=false --frozen-lockfile 2>&1
+
+  # prune the store after install to remove any package versions which may have been upgraded/removed
+  # since the last install.
+  suppress_output pnpm store prune
+}
+
+pnpm_prune_devdependencies() {
+  local build_dir=${1:-}
+
+  cd "$build_dir" || return
+
+  # seems like we can't use prune without some workarounds because it triggers pre/post-install scripts
+  # and does not respect the `--ignore-scripts` argument:
+  # https://github.com/pnpm/pnpm/issues/5030
+  #
+  # so we're going to create a temporary .npmrc file which does allow us to set the ignore-scripts config:
```

Comment:
While clever, this feels a bit brittle. Is pruning a necessity for our first pass at pnpm support on this buildpack? Just wondering if we want pruning bad enough to ship with a workaround instead of waiting for a fix upstream?

Reply:
I tried a couple of other approaches with mixed results in terms of outcome and output:
- just pruning (no `--ignore-scripts`) - this doesn't seem viable because if a script that depends on a dev dependency is triggered it will fail the build
- `rm -rf node_modules` and `pnpm install --prod` - this works but the output is confusing because it looks like a fresh install

I think I'd like to see this included in a first pass while working on (or waiting for) an upstream fix.  What do you think about an environment variable (like `SKIP_PNPM_PRUNING`) that could enable/disable this step in case it causes problems?

Reply:
Let's drop this .npmrc stuff. Maybe we only prune on versions of pnpm that support your `--ignore-scripts`?

Reply:
@joshwlewis this is implemented in https://github.com/heroku/heroku-buildpack-nodejs/pull/1238